pycbc/events/coinc.py

```diff
@@ -370,7 +370,97 @@ def cluster_coincs(stat, time1, time2, timeslide_id, slide, window, argmax=numpy
 
     span = (time.max() - time.min()) + window * 10
     time = time + span * tslide
+    cidx = cluster_given_mean_times(stat, time, window, argmax)
+    return cidx
+
+def cluster_coincs_multiifo(stat, time_coincs, timeslide_id, slide, window, argmax=numpy.argmax):
+    """Cluster coincident events for each timeslide separately, across
+    templates, based on the ranking statistic
+
+    Parameters
+    ----------
+    stat: numpy.ndarray
+        vector of ranking values to maximize
+    time_coincs: tuple of numpy.ndarrays
+        tuple containing time vectors each ifo
+    timeslide_id: numpy.ndarray
+        vector that determines the timeslide offset
+    slide: float
+        length of the timeslides offset interval
+    window: float
+        legth to cluster over
+
+    Returns
+    -------
+    cindex: numpy.ndarray
+        The set of indices corresponding to the surviving coincidences
+    """
+    time_coinc_zip = zip(*time_coincs)
+    if len(time_coinc_zip) == 0:
+        logging.info('No coincident triggers.')
+        return numpy.array([])
+
+    time = []
+    #find the mean of each time coincidences, providing that the IFO is on
+    for tc in time_coinc_zip:
+        time.append(mean_if_greater_than_zero(tc))
+
+    num_ifos = [len([t for t in tc if t > 0]) for tc in time_coinc_zip]
```

Comment:
think this could simplified by using a logical expression - eg `(tc > 0).sum() for tc in time_coinc_zip` ?

Reply:
This was sorted by the use of two outputs in mean_if_greater_than_zero anyway